src/util/charmrun-src/charmrun.C

```diff
@@ -2535,31 +2535,37 @@ static int req_handle_crash(ChMessage *msg, nodetab_process & p)
   /*Anounce crash to all child charmruns*/
   announce_crash(p);
 }
-
-#endif
 #endif
 
-#ifdef __FAULT__
-static void error_in_req_serve_client(nodetab_process & p)
+static void crashed_process_common(nodetab_process & p)
 {
-  const SOCKET fd = p.req_client;
-  fprintf(stderr, "Socket %d failed \n", fd);
+  skt_close(p.req_client);
 
-  fflush(stdout);
-  skt_close(fd);
+#if 0
+  // send a message to all the other processes telling them that this one has crashed
+  announce_crash(p);
+#endif
 
-  /** should also send a message to all the other processors telling them that
-   * this guy has crashed*/
-  /*announce_crash(p);*/
   restart_node(p);
 
-  fprintf(stderr, "charmrun says process %d failed (on host %s)\n", p.nodeno, p.host->name);
-  /** after the crashed processor has been recreated
-   it connects to charmrun. That data must now be filled
-   into my_process_table and the nodetab_table*/
-
+  // After the crashed processor has been recreated
+  // it connects to Charmrun. That data must now be filled
+  // into the node table.
   reconnect_crashed_client(p);
 }
+
+static void error_in_req_serve_client(nodetab_process & p)
+{
+  printf("Charmrun> Process %d (host %s) failed: Socket error\n", p.nodeno, p.host->name);
+  fflush(stdout);
+
+#if 0
```

Comment:
Why leave this in?

Reply:
Because maybe with more work it could be possible to "adequately distinguish host faults from process faults" and this is the place where it would be marked.

Reply:
Fair.